pyfar/dsp/fractional_octave_smoothing.py

```diff
@@ -0,0 +1,317 @@
+import numpy as np
+import cmath as cm
+from pyfar import Signal
+
+
+class FractionalSmoothing:
+    """ Class of fractional smoothing object.
+        Object contains data of a signal and a given smoothing width.
+    """
+    def __init__(
+            self,
+            n_bins,
+            smoothing_width):
+        """
+        Initiate FractionalSmoothing object.
+
+        Parameters
+        ----------
+        n_bins : int
+            Number of frequency bins of signal.
+        smoothing_width : float, int
+            Width of smoothing window relative to an octave
+
+        Raises
+        ------
+        TypeError
+            Invalid data type of smoothing_width (int/float).
+        TypeError
+            Invalid data type of n_bins (int).
+        """
+        if not isinstance(smoothing_width, (float, int)):
+            raise TypeError("Invalid data type of window width (int/float).")
+        if not isinstance(n_bins, int):
+            raise TypeError("Invalid data type of number of bins (int).")
+        # Get number of freq bins
+        self._n_bins = n_bins
+        # Save smoothing width:
+        self._smoothing_width = smoothing_width
+
+    def calc_integration_limits(self):
+        """
+        Computes integration limits for each frequency bin
+        k. For each k two arrays are computed, one containing the upper, one
```

Comment:
Start with single line doc-str:
```suggestion
        Compute integration limits for each frequency bin.
        
        For each k two arrays are computed, one containing the upper, one
```

You could also reference the equation number already in the docstring.

Reply:
Done. Also added equation references.